### ISSUE TITLE
replace Sage's closest vector by fpylll's code?

### ISSUE THREAD

**Issue by dimpase** [fplll/fpylll#176]
We now have to deal with Sage's IntegerLattice bugs on https://trac.sagemath.org/ticket/29866 - and
it appears that for some reason Sage does not use fpylll's code for closest_vector, but does something on its own (and that got broken in Sage 8.5, as it turned out).

On the example showing Sage's bug, fpylll does the job just fine. Would it be beneficial to replace Sage's code a code that just calls fpylll ?

**Comment by malb** [comment]
FP(y)LLL only permits integer target vectors, but Sage's current code handles rational target vectors too. This is mostly an interface issue but still.

**Comment by dimpase** [comment]
Does FP(y)LLL have code to compute the Voronoi cell? 
Further digging on the Sage ticket points at a mysterious breakage in [calculate_voronoi_cell()](https://github.com/sagemath/sage/blob/e2dcdeeabb578c37bcf0361c0be3079315e9252c/src/sage/modules/diamond_cutting.py#L240)

**Comment by malb** [comment]
Nope. Well, a sieving db is an approximate Voronoi cell so G6K could be used, but it would be heuristic and not provable.
